torchao/quantization/quantize_/workflows/intx/intx_unpacked_tensor.py

```diff
@@ -199,10 +216,18 @@ def _(func, types, args, kwargs):
         args[1],
         args[2] if len(args) > 2 else None,
     )
-    if isinstance(input_tensor, IntxUnpackedTensor):
-        input_tensor = input_tensor.dequantize()
-    if isinstance(weight_tensor, IntxUnpackedTensor):
-        weight_tensor = weight_tensor.dequantize()
+    assert isinstance(weight_tensor, IntxUnpackedTensor)
+
+    # Apply dynamic activation quant
+    if weight_tensor.apply_activation_quantization:
```

Comment:
also probably rename to `apply_int8_act_asym_per_token_quant` to be more specific probably